logstash-core/lib/logstash/util/wrapped_acked_queue.rb

```diff
@@ -196,29 +195,28 @@ def close_batch(batch)
         @mutex.lock
         begin
           batch.close
-          # there seems to be concurrency issues with metrics, keep it in the mutex
           @inflight_batches.delete(Thread.current)
-          stop_clock(batch)
         ensure
           @mutex.unlock
         end
+        stop_clock(batch)
       end
 
       def start_clock
         @inflight_clocks[Thread.current] = java.lang.System.current_time_millis
       end
 
       def stop_clock(batch)
-        unless @inflight_clocks[Thread.current].nil?
+        start_time = @inflight_clocks.get_and_set(Thread.current, nil)
+        unless start_time.nil?
```

Comment:
shouldn't it be an error of some kind that stop_clock is called without a prior start_clock?

Reply:
@jordansissel yea, actually it looks like this never happens (I can['t see a code path that would cause this in any way).
I'll investigate this soon :) Looks like a freebie performance improvement if this is actually never `nil` :) But I'd not mix that change in here right?